File: src/main/java/com/caicai/swe/entity/Product.java
package com.caicai.swe.entity;

import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 商品实体
 */
@Data
public class Product implements Serializable {
    private static final long serialVersionUID = 1L;

    private Long id;
    private Long userId;
    private String title;
    private String description;
    private BigDecimal price;
    private Long categoryId;
    private String images;
    private Integer status;
    private Integer viewCount;
    private Integer favoriteCount;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}

File: frontend/src/types/product.ts
// 商品相关类型
export interface ProductVO {
  id: number
  userId: number
  username: string
  userAvatar: string | null
  title: string
  description: string | null
  price: number
  categoryId: number | null
  categoryName: string | null
  images: string | null
  status: number
  viewCount: number
  favoriteCount: number
  createTime: string
}

export interface ProductPublishDTO {
  title: string
  description?: string
  price: number
  categoryId?: number
  images?: string
}

export interface ProductQueryDTO {
  page: number
  size: number
  keyword?: string
  categoryId?: number
  status?: number
  userId?: number
}